### PR TITLE
Responsive size listener performance

### PR DESCRIPTION
**Pull request by jnatalzia**
### This PR will...
Improve performance of the responsive resize listener

### Why is this Pull Request needed?
It performs poorly, especially with multiple players on page

### Are there any points in the code the reviewer needs to double check?
None I can think of

### Are there any Pull Requests open in other repos which need to be merged with this?
No

#### Addresses Issue(s):

JW8-2383

### CODE REVIEW

File: src/js/view/utils/resize-listener.js [hunk 1]
```diff
@@ -49,26 +54,29 @@ export default class ResizeListener {
...
                 this.callback(e, currentWidth);
             });
-            this.resizeRaf = resizeRaf;
             this.resetTriggers();
         };
 
-        element.addEventListener('scroll', this.scrollListener, true);
-        this.resetTriggers();
+        requestAnimationFrame(() => {
+            // TODO: this should be moved to the widget. The player element is always relative.
```
Comment (jnatalzia) [inline src/js/view/utils/resize-listener.js]: assume you mean shelf widget here @robwalch ?

Comment (robwalch) [reply src/js/view/utils/resize-listener.js]: Yep. The player must be styled `position: relative` or the absolute positioned elements contained within will fill the page: https://github.com/jwplayer/jwplayer/blob/master/src/css/jwplayer/imports/jwplayerlayout.less#L6

File: src/js/view/view.js [hunk 2]
```diff
@@ -341,6 +338,9 @@ function View(_api, _model) {
         // Triggering 'resize' resulting in player 'ready'
         _lastWidth = _lastHeight = null;
         this.checkResized();
+        if (!this.resizeListener) {
+            this.resizeListener = new ResizeListener(_playerElement, viewsManager.scheduleResponsiveRedraw);
```
Comment (robwalch) [inline src/js/view/view.js]: The most optimal way to fit this into view setup would be for this classes constructor to only do the DOM creation and styling - that can happen in `view.setup`. And then defer the reads until here (or just after `updateBounds` above).

File: test/unit/setup-test.js [hunk 3]
```diff
@@ -31,18 +33,25 @@ describe('api.setup', function() {
...
+            api.remove();
+        }
         if (fixture.parentNode) {
             fixture.parentNode.removeChild(fixture);
         }
         for (let i = instances.length; i--;) {
             instances[i].remove();
         }
         sandbox.restore();
+        api = null;
```
Comment (robwalch) [inline test/unit/setup-test.js]: Before these changes, were were not tearing down all api instances (and their views) properly, which left `view` instances in the `viewManager`s list of views.

**Comment by johnBartos** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  🔎 Assign some reviewers or assignees.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/js/view/view.js [hunk 4]
```diff
@@ -1013,6 +1009,7 @@ function View(_api, _model) {
         clearCss(_model.get('id'));
         if (this.resizeListener) {
             this.resizeListener.destroy();
+            delete this.resizeListener;
```
Comment (robwalch) [inline src/js/view/view.js]: we usually null these on destroy

**Pull request merged**